tools/kind-stateful-set.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: stateful
+  namespace: stateful-staging
+  labels:
+    app: nginx
+spec:
+  ports:
+  - port: 80
+    name: web
+  clusterIP: None
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: stateful
+  namespace: stateful-staging
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  serviceName: "nginx"
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+      - name: nginx
```

Comment:
just curious why nginx?

Reply:
Just the upstream statefulset example I stole from

Reply:
It also touches on all the components of a real world statefulset i.e a Headless Service used to control the network domain & the StatefulSet with the nginx container will launch Pods.

Reply:
If you'd prefer we can just have all our k8s resource run busybox as the container image

Reply:
i have no preference, @mcutalo88 or @scarlettperry can chime in if they feel strongly

Reply:
fine with me 👍